portal/src/components/FilterCategories/FilterCategories.component.js

```diff
@@ -40,72 +40,50 @@ export default {
       category.showAll = !category.showAll
       this.$forceUpdate()
     },
-    /**
-     * Set filter for v-model
-     * @returns {*} - filters
-     */
-    setFilter() {
-      const { categories, publisherdate } = this
-      if (categories) {
-        let filters = []
-
-        const publisherdate_item = this.value.filters.find(
-          item => item.external_id === publisherdate
-        )
-
-        if (publisherdate_item) {
-          filters.push({
-            external_id: publisherdate,
-            items: [...publisherdate_item.items]
-          })
-          this.data = {
-            start_date: publisherdate_item.items[0],
-            end_date: publisherdate_item.items[1]
-          }
-        }
-
-        filters = Object.assign({}, this.value, {
-          filters
-        })
-
-        this.$router.push(this.generateSearchMaterialsQuery(filters))
-
-        this.$emit('input', filters)
+    onChange(e) {
+      const { categoryId, itemId } = e.target.dataset
+      const filter = { external_id: categoryId, items: [] }
+      const categoryFilter =
+        this.selectedFilters.find(f => f.external_id === categoryId) || filter
+      const selectedFilters = this.selectedFilters.filter(
+        f => f.external_id !== categoryId
+      )
 
-        return filters
+      if (e.target.checked) {
+        categoryFilter.items = [...categoryFilter.items, itemId]
+      } else {
+        categoryFilter.items = categoryFilter.items.filter(f => f !== itemId)
       }
+      return this.executeSearch([...selectedFilters, categoryFilter])
     },
-    onChange(event, parent) {
-      if (!_.isNil(parent)) {
-        parent.selected = !parent.selected
+    onDateChange(dates) {
+      const { start_date, end_date } = dates
+      const filter = {
+        external_id: this.publisherDateExternalId,
+        items: [start_date, end_date]
       }
-      this.$store.commit('SET_FILTER_SELECTED', event.target.dataset.categoryId)
-      this.executeSearch()
-    },
-    onDateChange() {
-      this.executeSearch()
+      const filters = this.selectedFilters.filter(
```

Comment:
Ik zou een map doen, waar we altijd het origineel returnen, tenzij het die publisherdate is.

Reply:
kun je me uitleggen waarom? volgens mij wordt het er alleen maar ingewikkelder op

Reply:
Ja, het kan er natuurlijk ook niet in zetten (dat had ik even over het hoofd gezeten). Dan zou ik er eerder een object met key/values van maken. Dan kun je de key zetten met de nieuwe values (en als het al bestaat wordt het overschreven). Maar dat is weer een hele refactor van bestaande code, dus laat maar zitten.